### PR TITLE
add time functions

### PR DESCRIPTION
**Pull request by astrophysik**
General
-----------
Add several time builtins. Functions `iso_week_number, date, fix_year` were toked from `datetime_functions.cpp` with few modifications. In the future, this should be moved to common

### CODE REVIEW

File: runtime-light/stdlib/time/time-functions.cpp [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+  return duration_cast<std::chrono::seconds>(now).count();
+}
+
+int64_t f$mktime(int64_t hour, Optional<int64_t> minute, Optional<int64_t> second, Optional<int64_t> month, Optional<int64_t> day,
+                 Optional<int64_t> year) noexcept {
+  using namespace std::chrono;
+  auto time_since_epoch{system_clock::now().time_since_epoch()};
+  year_month_day current_date{sys_days{duration_cast<std::chrono::days>(time_since_epoch)}};
+
+  auto hours{std::chrono::hours(hour)};
```
[the remark this answers is not in the document]
Comment (astrophysik) [reply runtime-light/stdlib/time/time-functions.cpp:380]: Actually, I use it to simplify access to `system_clock, year_month_day, sys_days` etc. I use full name only for time units

File: runtime-light/stdlib/time/time-functions.cpp [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+constexpr std::array<std::string_view, 12> PHP_TIMELIB_MON_FULL_NAMES = {"January", "February", "March",     "April",   "May",      "June",
+                                                                         "July",    "August",   "September", "October", "November", "December"};
+constexpr std::array<std::string_view, 12> PHP_TIMELIB_MON_SHORT_NAMES = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
+constexpr std::array<std::string_view, 7> PHP_TIMELIB_DAY_FULL_NAMES = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+constexpr std::array<std::string_view, 7> PHP_TIMELIB_DAY_SHORT_NAMES = {"Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"};
+
+constexpr std::array<std::string_view, 4> suffix = {"st", "nd", "rd", "th"};
+
+void iso_week_number(int y, int doy, int weekday, int &iw, int &iy) {
```
[the remark this answers is not in the document]
Comment (astrophysik) [reply runtime-light/stdlib/time/time-functions.cpp]: fix

File: runtime-light/stdlib/time/time-functions.cpp [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+    int j;
+    j = doy + (7 - weekday) + jan1weekday;
+    iw = j / 7;
+    if (jan1weekday > 4) {
+      iw -= 1;
+    }
+  }
+}
+
+string date(const string &format, const tm &t, int64_t timestamp, bool local) {
```
[the remark this answers is not in the document]
Comment (astrophysik) [reply runtime-light/stdlib/time/time-functions.cpp]: fix

File: runtime-light/stdlib/time/time-functions.cpp [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+        }
+        /* fallthrough */
+      default:
+        SB << format[i];
+    }
+  }
+  return SB.str();
+}
+
+inline int32_t fix_year(int64_t year) {
```
[the remark this answers is not in the document]
Comment (astrophysik) [reply runtime-light/stdlib/time/time-functions.cpp]: fix

File: runtime-light/stdlib/time/time-functions.cpp [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+#include <ctime>
+#include <string_view>
+
+#include "runtime-common/core/runtime-core.h"
+
+namespace {
+constexpr std::string_view PHP_TIMELIB_TZ_MOSCOW = "Europe/Moscow";
+constexpr std::string_view PHP_TIMELIB_TZ_GMT3 = "Etc/GMT-3";
+
+constexpr std::array<std::string_view, 12> PHP_TIMELIB_MON_FULL_NAMES = {"January", "February", "March",     "April",   "May",      "June",
```
[the remark this answers is not in the document]
Comment (astrophysik) [reply runtime-light/stdlib/time/time-functions.cpp:19]: It isn't compile without size

**Review by apolyakov — DISMISSED** [review]

**Review by apolyakov — APPROVED** [review]

**Review by apolyakov — APPROVED** [review]

**Pull request merged**